subsys/net/lib/dns/resolve.c

```diff
@@ -1422,6 +1423,10 @@ int dns_resolve_reconfigure(struct dns_resolve_context *ctx,
 
 struct dns_resolve_context *dns_resolve_get_default(void)
 {
+	if (!default_ctx_initialized) {
+		dns_resolve_init(&dns_default_ctx, NULL, NULL);
+		default_ctx_initialized = true;
+	}
```

Comment:
Hmm, the `dns_resolve_init()` is normally called just fine during system initialization by `dns_init_resolver()` in `subsys/net/ip/net_core.c`. But that DNS init might be skipped if DHCPv4 is not properly initialized, I suspect that something like this happens here.
The `dns_resolve_init()` in `dns_init_resolver()` looks like this
```
        ret = dns_resolve_init(dns_resolve_get_default(), dns_servers, NULL);
        if (ret < 0) {
                NET_WARN("Cannot initialize DNS resolver (%d)", ret);
        }
```
It looks like there would be issues here if we do it like this.

Reply:
What's the proper way to fix this, then?  You can see this pretty trivially in native_posix with tests/net/dhcpv4, just run it under gdb and set a breakpoint right here and on dns_resolve_init() to see that the latter never gets called.

Reply:
@andyross I just submitted a fix proposal at https://github.com/zephyrproject-rtos/zephyr/pull/51267
You could just take the fix commit from there or wait until until it gets merged.